internal/juju/secrets.go

```diff
@@ -134,8 +135,12 @@ func (c *secretsClient) DeleteSecret(input *DeleteSecretInput) error {
 	}
 
 	secretAPIClient := c.getSecretAPIClient(conn)
+	secretURI, err := coresecrets.ParseURI(input.SecretId)
+	if err != nil {
+		return err
+	}
 	// TODO: think about removing concrete revision.
-	err = secretAPIClient.RemoveSecret(nil, input.Name, nil)
+	err = secretAPIClient.RemoveSecret(secretURI, input.Name, nil)
```

Comment:
Supplying both the secertURI and the name seems to be a problem for all calls but Create: https://github.com/juju/juju/blob/63c7f8db6dac0995d92c5513d499a01c6c755d4c/apiserver/facades/client/secrets/secrets.go#L428

Please check the others.